### ISSUE TITLE
GitHub release

### ISSUE THREAD

**Issue by baimafeima** [proxysh/Safejumper-for-Desktop#1]
I just installed your .deb package in Ubuntu. Could you please provide versioned tarballs here on GitHub to allow for a more distro-agnostic use of your software?

**Comment by proxysh** [comment]
Do you mean the end binaries? You can find them at https://proxy.sh/software :)

**Comment by baimafeima** [comment]
You only provide a .zip file there as far as I can see. It would be great to have versioned tarball releases here on GitHub. This would also make it much easier checking for new releases and allow packaging for various Linux-based distributions.

**Comment by proxysh** [comment]
Please check inside the zip, you have tarballs and packages. All our Linux binaries will be archived into a single zip for proper distribution.

**Comment by baimafeima** [comment]
I really prefer these to be distributed separately and on GitHub. This would make it easier to checksum and verify and increase distribution security. I also won't be notified timely about new versions when there are no releases here on GitHub.

**Comment by proxysh** [comment]
The checksum is verified for the .zip so this is all good. And distributing across multiple channels does not seem like a wise solution that will provide any security value. We will stick to a single distribution channel: proxy.sh/linux, pointing to a safejumper_linux.zip archive with proper checksum verification. Within that zip, you will find the binaries et. al. Another reason we do that, is that in our next version, we will cover more distros than the current build 60, so we will have to deliver new binaries in addition to those already provided in the zip.

**Comment by baimafeima** [comment]
> Within that zip, you will find the binaries et. al. Another reason we do that, is that in our next version, we will cover more distros than the current build 60, so we will have to deliver new binaries in addition to those already provided in the zip.

This has the disadvantage that I have to download files which I actually do not need. The .zip file is also not versioned and I do not immediately see what version it is that I am downloading. I don't think it will be much more work on your side to release these files separately on GitHub.

**Comment by proxysh** [comment]
I understand, but the zip is constantly evolving, including more various files for more distros, so it's really unnecessary hard work to have to release independent binaries and sign them all. I hope you will not mind coping with a single download file. Let us know once again if this is really a heart breaker for you. Thanks.

**Comment by baimafeima** [comment]
> Let us know once again if this is really a heart breaker for you. 

I have suggested Safejumper for inclusion into the [Solus](https://solus-project.com/) repository but it is unlikely that it can be added and maintained given the current release situation. 

I know you are an extremely responsive support team and I'd appreciate your contributions on the dev tracker. This will also make it easier to discuss many of these technical aspects directly with the Solus developers. You can find their bug and development tracker here:
https://dev.solus-project.com/T2428

Four more reasons why a GitHub release is important: 
- easy way to make source code available
- offer a reliable download source as GitHub is not blocked in censorship-affected regions
- versioned releases to quickly identify whether an update is available
- separate releases for quicker downloads (there is really no point in downloading an entire zip file with files I don't need, especially in regions where bandwidth is crucially limited)

**Comment by proxysh** [comment]
Are you suggesting that we should push all binaries on Github as well?

As for Solus, we cannot really weigh in, it does not really add any value to their OS to see Safejumper included as a default package.

**Comment by baimafeima** [comment]
> Are you suggesting that we should push all binaries on Github as well?

The source packages are most important but I wouldn't mind having the binaries available on GitHub as well. The .deb package currently doesn't work on Ubuntu MATE, by the way. The application crashes, so you might look into this as well. 

Regarding Solus, I merely suggested it for inclusion but the decision to include it is up to them. In any case, it won't be a default package to be found on an ISO, just an optional package to be distributed via their repository. Thank you again for reconsidering your release policy.

**Comment by proxysh** [comment]
The sources are already provided here on Github. Our source code is full and open source. Could you please elaborate?

**Comment by baimafeima** [comment]
Only downloadable .zip files. As I mentioned, there are no versioned tarballs or binaries in the release section.

**Comment by baimafeima** [comment]
> Could you please elaborate?

The biggest problems I have with your release policy is: 
1. You don't offer GitHub releases. As I have mentioned, GitHub is easily accessible in censorship-affected areas. Governments usually don't block GitHub as this would be disastrous for the development community. Your website, on the other hand, is difficult to access. Your use of Google Plugins is problematic. Why do you need these plugins in the first place when you have a zero log policy? In case your website is censored, a GitHub release would at least offer an alternative way to download updates. 
2. You don't offer versioned releases via your website. This is unreasonable as I will only ḱnow _after_ downloading what files I have downloaded and whether there were any updates available. Why don't you simply add a date to the zip file? 
3. You don't offer separate downloads. Bundled zip files take forever to download. Could you please explain to me why I should download something which I do not need? I also don't go to a store and buy stuff only to throw it away when I am back home.
![sj](https://cloud.githubusercontent.com/assets/23112309/23101662/29390cdc-f699-11e6-8409-396575145b5c.png)

Just for your info: I have to cancel this download. It is not reasonable to wait for over an hour. With separate releases here on GitHub this is a matter of minutes.

**Comment by proxysh** [comment]
Your feedback makes sense, but it is a nightmare for us to organise it that way, signing all files and having to upload across various CDNs. We will think of a better distribution according to your feedback, anyway. Thanks again for your precious time.

**Comment by baimafeima** [comment]
Thanks for being open to feedback.

**Comment by proxysh** [comment]
No problem, anytime.

**Comment by proxysh** [comment]
We've investigated and we're unable to publish binaries to Github because they limit filesizes to 25MB. If you want, we can upload mirrors to some file-hosting sites, just let us know where. Meanwhile, we've improved our versioning and PGP keys are available directly under proxy.sh/software thanks to your feedback and that of others :)

**Comment by baimafeima** [comment]
Thanks a lot for improving your release policy. Much appreciated!

**Comment by proxysh** [comment]
Glad to hear you like it! :)

**Comment by baimafeima** [comment]
> We've investigated and we're unable to publish binaries to Github because they limit filesizes to 25MB.

Allright, I am sure there are options to upgrade your account but more importantly, you could release versioned tarballs here. These should not take up too much space. 

I have another problem with your current release policy. I don't understand why you indicate the build number on your website. Usually this kind of information is part of the actual release file. The point of having a file versioned is to quickly identify what it is that you are downloading. Today, I was notified via the Safejumper app that build 75 is available but this is not indicated on your website. Again, the only option I have is to download a 175 MB zip file. This is not an accepted form of releasing software.

**Comment by AlkalineJaunt** [comment]
Further, I can not locate any release notes, and your commit history is all but absent. Why is this?

And please; version your archives.

**Comment by jpwhiting** [comment]
Ah, I forgot to push to the public repository for the past few builds. Doing so now.

**Comment by AlkalineJaunt** [comment]
I will say this: You are very responsive. Thanks.

**Comment by baimafeima** [comment]
Thanks a lot!

**Comment by proxysh** [comment]
No problem guys. The version number is available on our site (https://proxy.sh/software & https://proxy.sh/client). The PGP key is accurate. All is now good.

There were two suggestions you made but we did not implement: Github release (impossible due to filesize issue) and splitting the Linux archives into various binaries (too much time-consuming for each release, especially if we have to sign all individual files each time).

We hope you appreciate those moves and the decisions we took. Enjoy!
